configure.ac

```diff
@@ -237,6 +237,14 @@ AC_ARG_ENABLE([extra-example-tests],
 test "x$enable_extra_example_tests" = xyes || enable_extra_example_tests=no
 AC_MSG_RESULT($enable_extra_example_tests)
 
+# Does the user want to run filter example
+AC_MSG_CHECKING([whether filter example should be run])
+AC_ARG_ENABLE([filter-example],
+              [AS_HELP_STRING([--enable-filter-example],
+                              [Run filter example; requires netCDF-4])])
+test "x$enable_filter_example" = xyes || enable_filter_example=no
+AC_MSG_RESULT($enable_filter_example)
+
```

Comment:
Why not just run the example every time. like the other examples? And if the user really objects, let them use disable-examples to turn them all off.

A proliferation of configure options makes testing more challenging. Ideally, I would double my CI test cases every time someone adds an --enable/--disable to configure. In practice, I don't do that, but I still like to add some tests for every option.

So adding an option should only be done when there is a good reason why some users would use it. I don't see any good reason to use --disable-filter-example. I suggest you simply remove it, and let the example get built every time, unless there is some good reason why it should not be run sometimes.